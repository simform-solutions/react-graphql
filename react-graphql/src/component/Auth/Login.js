import React from "react";
import { Formik } from "formik";
import * as yup from "yup";
import { graphql } from "react-apollo";
import gql from "graphql-tag";
import { compose, withState, withHandlers } from "recompose";
import { withRouter } from "react-router-dom";

const fields = [
  {
    label: "Email",
    name: "email",
    type: "text"
  },
  {
    label: "Password",
    name: "password",
    type: "password"
  }
];
const validation = {
  email: yup
    .string()
    .email("E-mail is not valid!")
    .required("E-mail is required!"),
  password: yup
    .string()
    .min(5)
    .required("Password is required!")
};

const initialValues = {
  email: "",
  password: ""
};

const LOGIN = gql`
  mutation SubmitForm($email: String!, $password: String!) {
    login(email: $email, password: $password) {
      email
      firstName
      lastName,
      token,
      message
    }
  }
`;

export const LoginForm = ({
  values,
  touched,
  errors,
  isSubmitting,
  handleChange,
  handleBlur,
  handleSubmit,
  handleReset
}) => (
  <form onSubmit={handleSubmit}>
    {fields.map(({ label, name, type }) => {
      return (
        <div key={name} className="form-group row">
          <label className="col-sm-2 col-form-label">{label}</label>
          <div className="col-sm-10">
            <input
              name={name}
              type={type}
              className={`form-control ${errors[name] &&
                touched[name] &&
                "is-invalid"}`}
              value={values[name]}
              onChange={handleChange}
              onBlur={handleBlur}
            />
            {errors[name] &&
              touched[name] && (
                <div className="invalid-feedback">{errors[name]}</div>
              )}
          </div>
        </div>
      );
    })}
    <button
      disabled={isSubmitting}
      type="submit"
      className="btn btn-primary"
      style={{ marginRight: "10px" }}
    >
      Login
    </button>
    <button type="button" className="btn btn-danger" onClick={handleReset}>
      Reset
    </button>
  </form>
);

export const Login = ({ state, onFormSubmit }) => (
  <div>
    <h1>Login</h1>
    <Formik
      initialValues={initialValues}
      validationSchema={yup.object().shape(validation)}
      onSubmit={onFormSubmit}
      component={LoginForm}
    />
    {state &&
      state.errorMessage && (
        <div className="snackbar show red">{state.errorMessage}</div>
      )}
  </div>
);

export default compose(
  withState("state", "setState", {
    errorMessage: ""
  }),
  graphql(LOGIN),
  withRouter,
  withHandlers({
    onFormSubmit: ({ mutate, setState, history }) => async (
      values,
      { setSubmitting, resetForm }
    ) => {
      try {
        const  {data: {
          login: { firstName, lastName,token, message }
        }} = await mutate({
          variables: values
        });
        localStorage.setItem("session-token", token);
        localStorage.setItem("message", message);
        setSubmitting(false);
        if (firstName) {
          resetForm(initialValues);
          history.push("/");
        } else {
          setState(state => ({
            ...state,
            errorMessage: "Invalid Credentials"
          }));
          setTimeout(() => {
            setState(state => ({
              ...state,
              errorMessage: ""
            }));
          }, 3000);
        }
      } catch (err) {
        setSubmitting(false);
        setState(state => ({
          ...state,
          errorMessage: err.message
        }));
        setTimeout(() => {
          setState(state => ({
            ...state,
            errorMessage: ""
          }));
        }, 3000);
      }
    }
  })
)(Login);
