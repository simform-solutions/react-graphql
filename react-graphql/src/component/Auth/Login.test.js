import React from "react";

import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

import { Login, LoginForm } from "./Login";
const loginFormProps = {
  values: {
    email: "",
    password: ""
  },
  errors: {},
  touched: {},
  isSubmitting: false,
  isValidating: false,
  submitCount: 0,
  dirty: false,
  isValid: false,
  initialValues: {
    email: "",
    password: ""
  },
  validateOnChange: true,
  validateOnBlur: true
};

describe("Login", () => {
  it("Login should renders without crashing", () => {
    const component = shallow(<Login />);
    expect(component).toMatchSnapshot();
  });

  it("Login should contains title", () => {
    const component = shallow(<Login />);
    const title = <h1>Login</h1>;
    expect(component.contains(title)).toEqual(true);
  });

  it("Login should contains a Formik Tag", () => {
    const component = shallow(<Login />);
    expect(component.find("Formik")).toHaveLength(1);
  });

  it("Login should show error message", () => {
    const component = shallow(<Login state={{ errorMessage: "error!" }} />);
    expect(component.find(".snackbar.red")).toHaveLength(1);
  });

  it("LoginForm should renders without crashing", () => {
    const component = shallow(<LoginForm {...loginFormProps} />);
    expect(component).toMatchSnapshot();
  });

  it("LoginForm should contains a form tag", () => {
    const component = shallow(<LoginForm {...loginFormProps} />);
    expect(component.find("form")).toHaveLength(1);
  });

  it("LoginForm should contains two buttons", () => {
    const component = shallow(<LoginForm {...loginFormProps} />);
    expect(component.find("button")).toHaveLength(2);
  });

  it("LoginForm submit button disable when isSubmitting is true", () => {
    const component = shallow(<LoginForm {...loginFormProps} isSubmitting />);
    expect(component.find("button[type='submit']").props().disabled).toEqual(
      true
    );
  });

  it("LoginForm should contains a password field ", () => {
    const component = shallow(<LoginForm {...loginFormProps} />);
    expect(component.find("input[name='password']")).toHaveLength(1);
  });

  it("LoginForm should contains a email field ", () => {
    const component = shallow(<LoginForm {...loginFormProps} />);
    expect(component.find("input[name='email']")).toHaveLength(1);
  });

  it("LoginForm submit button should have text Login", () => {
    const component = shallow(<LoginForm {...loginFormProps} />);
    expect(component.find("button[type='submit']").text()).toEqual("Login");
  });
  it("LoginForm reset button should have text Reset", () => {
    const component = shallow(<LoginForm {...loginFormProps} />);
    expect(component.find("button[type='button']").text()).toEqual("Reset");
  });
});
