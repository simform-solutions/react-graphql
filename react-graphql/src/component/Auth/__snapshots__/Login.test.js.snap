// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Login should renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Login
        </h1>,
        <Formik
          component={[Function]}
          enableReinitialize={false}
          initialValues={
            Object {
              "email": "",
              "password": "",
            }
          }
          isInitialValid={false}
          validateOnBlur={true}
          validateOnChange={true}
          validationSchema={
            ObjectSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_defaultDefault": [Function],
              "_deps": Array [],
              "_excludedEdges": Array [],
              "_exclusive": Object {},
              "_mutate": false,
              "_nodes": Array [
                "password",
                "email",
              ],
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "object",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "fields": Object {
                "email": StringSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "required": false,
                    "undefined": false,
                  },
                  "_mutate": false,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "string",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
                "password": StringSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "min": true,
                    "required": false,
                  },
                  "_mutate": false,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "string",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
              },
              "tests": Array [],
              "transforms": Array [
                [Function],
              ],
            }
          }
        />,
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login",
        },
        "ref": null,
        "rendered": "Login",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "enableReinitialize": false,
          "initialValues": Object {
            "email": "",
            "password": "",
          },
          "isInitialValid": false,
          "onSubmit": undefined,
          "validateOnBlur": true,
          "validateOnChange": true,
          "validationSchema": ObjectSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_defaultDefault": [Function],
            "_deps": Array [],
            "_excludedEdges": Array [],
            "_exclusive": Object {},
            "_mutate": false,
            "_nodes": Array [
              "password",
              "email",
            ],
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "object",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "fields": Object {
              "email": StringSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_deps": Array [],
                "_exclusive": Object {
                  "required": false,
                  "undefined": false,
                },
                "_mutate": false,
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "string",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "tests": Array [
                  [Function],
                  [Function],
                  [Function],
                ],
                "transforms": Array [
                  [Function],
                ],
              },
              "password": StringSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_deps": Array [],
                "_exclusive": Object {
                  "min": true,
                  "required": false,
                },
                "_mutate": false,
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "string",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "tests": Array [
                  [Function],
                  [Function],
                  [Function],
                ],
                "transforms": Array [
                  [Function],
                ],
              },
            },
            "tests": Array [],
            "transforms": Array [
              [Function],
            ],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Login
          </h1>,
          <Formik
            component={[Function]}
            enableReinitialize={false}
            initialValues={
              Object {
                "email": "",
                "password": "",
              }
            }
            isInitialValid={false}
            validateOnBlur={true}
            validateOnChange={true}
            validationSchema={
              ObjectSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_defaultDefault": [Function],
                "_deps": Array [],
                "_excludedEdges": Array [],
                "_exclusive": Object {},
                "_mutate": false,
                "_nodes": Array [
                  "password",
                  "email",
                ],
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "object",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "fields": Object {
                  "email": StringSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": false,
                      "undefined": false,
                    },
                    "_mutate": false,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "string",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "tests": Array [
                      [Function],
                      [Function],
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                  },
                  "password": StringSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "min": true,
                      "required": false,
                    },
                    "_mutate": false,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "string",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "tests": Array [
                      [Function],
                      [Function],
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                  },
                },
                "tests": Array [],
                "transforms": Array [
                  [Function],
                ],
              }
            }
          />,
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
          },
          "ref": null,
          "rendered": "Login",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "enableReinitialize": false,
            "initialValues": Object {
              "email": "",
              "password": "",
            },
            "isInitialValid": false,
            "onSubmit": undefined,
            "validateOnBlur": true,
            "validateOnChange": true,
            "validationSchema": ObjectSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_defaultDefault": [Function],
              "_deps": Array [],
              "_excludedEdges": Array [],
              "_exclusive": Object {},
              "_mutate": false,
              "_nodes": Array [
                "password",
                "email",
              ],
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "object",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "fields": Object {
                "email": StringSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "required": false,
                    "undefined": false,
                  },
                  "_mutate": false,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "string",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
                "password": StringSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "min": true,
                    "required": false,
                  },
                  "_mutate": false,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "string",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
              },
              "tests": Array [],
              "transforms": Array [
                [Function],
              ],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Login LoginForm should renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginForm
    dirty={false}
    errors={Object {}}
    initialValues={
      Object {
        "email": "",
        "password": "",
      }
    }
    isSubmitting={false}
    isValid={false}
    isValidating={false}
    submitCount={0}
    touched={Object {}}
    validateOnBlur={true}
    validateOnChange={true}
    values={
      Object {
        "email": "",
        "password": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <div
            className="form-group row"
          >
            <label
              className="col-sm-2 col-form-label"
            >
              Email
            </label>
            <div
              className="col-sm-10"
            >
              <input
                className="form-control undefined"
                name="email"
                type="text"
                value=""
              />
            </div>
          </div>,
          <div
            className="form-group row"
          >
            <label
              className="col-sm-2 col-form-label"
            >
              Password
            </label>
            <div
              className="col-sm-10"
            >
              <input
                className="form-control undefined"
                name="password"
                type="password"
                value=""
              />
            </div>
          </div>,
        ],
        <button
          className="btn btn-primary"
          disabled={false}
          style={
            Object {
              "marginRight": "10px",
            }
          }
          type="submit"
        >
          Login
        </button>,
        <button
          className="btn btn-danger"
          type="button"
        >
          Reset
        </button>,
      ],
      "onSubmit": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "email",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="col-sm-2 col-form-label"
            >
              Email
            </label>,
            <div
              className="col-sm-10"
            >
              <input
                className="form-control undefined"
                name="email"
                type="text"
                value=""
              />
            </div>,
          ],
          "className": "form-group row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Email",
              "className": "col-sm-2 col-form-label",
            },
            "ref": null,
            "rendered": "Email",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="form-control undefined"
                  name="email"
                  type="text"
                  value=""
                />,
                undefined,
              ],
              "className": "col-sm-10",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "form-control undefined",
                  "name": "email",
                  "onBlur": undefined,
                  "onChange": undefined,
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              undefined,
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "password",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="col-sm-2 col-form-label"
            >
              Password
            </label>,
            <div
              className="col-sm-10"
            >
              <input
                className="form-control undefined"
                name="password"
                type="password"
                value=""
              />
            </div>,
          ],
          "className": "form-group row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Password",
              "className": "col-sm-2 col-form-label",
            },
            "ref": null,
            "rendered": "Password",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="form-control undefined"
                  name="password"
                  type="password"
                  value=""
                />,
                undefined,
              ],
              "className": "col-sm-10",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "form-control undefined",
                  "name": "password",
                  "onBlur": undefined,
                  "onChange": undefined,
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              undefined,
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login",
          "className": "btn btn-primary",
          "disabled": false,
          "style": Object {
            "marginRight": "10px",
          },
          "type": "submit",
        },
        "ref": null,
        "rendered": "Login",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Reset",
          "className": "btn btn-danger",
          "onClick": undefined,
          "type": "button",
        },
        "ref": null,
        "rendered": "Reset",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <div
              className="form-group row"
            >
              <label
                className="col-sm-2 col-form-label"
              >
                Email
              </label>
              <div
                className="col-sm-10"
              >
                <input
                  className="form-control undefined"
                  name="email"
                  type="text"
                  value=""
                />
              </div>
            </div>,
            <div
              className="form-group row"
            >
              <label
                className="col-sm-2 col-form-label"
              >
                Password
              </label>
              <div
                className="col-sm-10"
              >
                <input
                  className="form-control undefined"
                  name="password"
                  type="password"
                  value=""
                />
              </div>
            </div>,
          ],
          <button
            className="btn btn-primary"
            disabled={false}
            style={
              Object {
                "marginRight": "10px",
              }
            }
            type="submit"
          >
            Login
          </button>,
          <button
            className="btn btn-danger"
            type="button"
          >
            Reset
          </button>,
        ],
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "email",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="col-sm-2 col-form-label"
              >
                Email
              </label>,
              <div
                className="col-sm-10"
              >
                <input
                  className="form-control undefined"
                  name="email"
                  type="text"
                  value=""
                />
              </div>,
            ],
            "className": "form-group row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Email",
                "className": "col-sm-2 col-form-label",
              },
              "ref": null,
              "rendered": "Email",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    className="form-control undefined"
                    name="email"
                    type="text"
                    value=""
                  />,
                  undefined,
                ],
                "className": "col-sm-10",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "form-control undefined",
                    "name": "email",
                    "onBlur": undefined,
                    "onChange": undefined,
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                undefined,
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "password",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="col-sm-2 col-form-label"
              >
                Password
              </label>,
              <div
                className="col-sm-10"
              >
                <input
                  className="form-control undefined"
                  name="password"
                  type="password"
                  value=""
                />
              </div>,
            ],
            "className": "form-group row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Password",
                "className": "col-sm-2 col-form-label",
              },
              "ref": null,
              "rendered": "Password",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    className="form-control undefined"
                    name="password"
                    type="password"
                    value=""
                  />,
                  undefined,
                ],
                "className": "col-sm-10",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "form-control undefined",
                    "name": "password",
                    "onBlur": undefined,
                    "onChange": undefined,
                    "type": "password",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                undefined,
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
            "className": "btn btn-primary",
            "disabled": false,
            "style": Object {
              "marginRight": "10px",
            },
            "type": "submit",
          },
          "ref": null,
          "rendered": "Login",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Reset",
            "className": "btn btn-danger",
            "onClick": undefined,
            "type": "button",
          },
          "ref": null,
          "rendered": "Reset",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
